From 103b09e0d104db1b6842f45e1229f4fe81bec938 Mon Sep 17 00:00:00 2001
From: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
Date: Wed, 13 Feb 2019 15:49:19 +0100
Subject: zlib: GCC: Disable CRC32 optimizations on ARM

Starting with M70 and https://chromium-review.googlesource.com/c/1147867,
the code which leverages ARMv8 instructions to speed up CRC32 calculations
does not build correctly with GCC.

The second hunk that removes the "is_clang" condition is necessary
nonetheless, as the files included in the "zlib_arm_crc32" source set are
always required.

diff --git a/third_party/zlib/BUILD.gn b/third_party/zlib/BUILD.gn
index 51d477a331b8..3b65a40bb54a 100644
--- a/third_party/zlib/BUILD.gn
+++ b/third_party/zlib/BUILD.gn
@@ -68,7 +68,7 @@ if (use_arm_neon_optimizations) {
     #  - Fuchsia just added a syscall for feature detection.
     # TODO(cavalcantii): crbug.com/810125.
     if (!is_ios && !is_fuchsia) {
-      defines = [ "CRC32_ARMV8_CRC32" ]
+      defines = []
       if (is_android) {
         defines += [ "ARMV8_OS_ANDROID" ]
       } else if (is_linux || is_chromeos) {
@@ -80,7 +80,7 @@ if (use_arm_neon_optimizations) {
   source_set("zlib_arm_crc32") {
     visibility = [ ":*" ]
 
-    if (is_clang && (!is_ios && !is_fuchsia)) {
+    if (!is_ios && !is_fuchsia) {
       include_dirs = [ "." ]
 
       if (is_android) {
-- 
2.17.1

